File: demo-service-impl/src/main/java/com/github/zubmike/service/demo/dao/ZoneSpaceDao.java
package com.github.zubmike.service.demo.dao;

import com.github.zubmike.core.dao.EntityItemDao;
import com.github.zubmike.service.demo.types.ZoneSpace;

import java.util.List;
import java.util.Optional;

public interface ZoneSpaceDao extends EntityItemDao<Long, ZoneSpace> {

	List<ZoneSpace> getAllByZone(int zoneId);

	Optional<ZoneSpace> getByStarship(long starshipId);

	Optional<ZoneSpace> getUsedSpace(int zoneId, long starshipId);

}

File: demo-service-kotlin-impl/src/main/kotlin/com/github/zubmike/service/demo/dao/ZoneSpaceDao.kt
package com.github.zubmike.service.demo.dao

import com.github.zubmike.core.dao.EntityItemDao
import com.github.zubmike.service.demo.types.ZoneSpace

interface ZoneSpaceDao : EntityItemDao<Long, ZoneSpace> {

    fun getAllByZone(zoneId: Int): List<ZoneSpace>

    fun getByStarship(starshipId: Long): ZoneSpace?

    fun getUsedSpace(zoneId: Int, starshipId: Long): ZoneSpace?

}
